libraries/AP_HAL/utility/RCOutput_Tap_Linux.cpp

```diff
@@ -23,6 +23,7 @@
 #include <asm/termbits.h>
 #include <string.h>
 #include <sys/ioctl.h>
+#include <asm/ioctls.h>
```

Comment:
I'm not sure this is correct.  Does your toolchain have the kernel headers? Because IOCTL numbers may be different from one platform to another and here I think you are actually going to use the numbers from your host, not the target.

Please check asm/ioctls.h being included here is actually coming from the toolchain and not from the host.  Some old thread I found about that: https://www.openwall.com/lists/musl/2017/11/23/2

Reply:
I can confirm that is using the file provided by the musl toolchain, a simple pragma did the job:
```
[ 385/1117] Compiling libraries/AP_HAL/Storage.cpp                                                                                                                                                                   
In file included from /home/patrick/toolchains/arm-linux-musleabihf-cross/arm-linux-musleabihf/include/linux/ioctl.h:5,                                                                                              
                 from /home/patrick/toolchains/arm-linux-musleabihf-cross/arm-linux-musleabihf/include/asm-generic/ioctls.h:5,                                                                                       
                 from /home/patrick/toolchains/arm-linux-musleabihf-cross/arm-linux-musleabihf/include/asm/ioctls.h:7,                                                                                               
                 from ../../libraries/AP_HAL/utility/RCOutput_Tap_Linux.cpp:26:                                                                                                                                      
/home/patrick/toolchains/arm-linux-musleabihf-cross/arm-linux-musleabihf/include/asm/ioctl.h:2:17: note: #pragma message: using musl asm/ioctl.h                                                                     
    2 | #pragma message "using musl asm/ioctl.h"                                                                                                                                                                     
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~
```

musl provides the same files as gnueabi, the only difference is that `bits/ioctl.h` (used by `sys/ioctl.h`) from musl does not include `asm/ioctls.h` as gnueabi do.

An alternative is to include `linux/ioctl.h` in the place of `asm/ioctls.h`, but I'm not sure if that is the appropriate approach.